### PR TITLE
Decksharing Tooltip enabled

### LINKED ISSUE #942: Accessibility: Support accessibility for Deck Sharing screen
Make sure that accessibility is supported for screen:
`delern/flutter/lib/views/deck_sharing/deck_sharing.dart`
      
       

**Doc**: https://flutter.dev/docs/development/accessibility-and-localization/accessibility
**Tooltip - Widget of the week**: https://www.youtube.com/watch?v=EeEfD5fI-5Q

### PR DESCRIPTION
**Pull request by jddeep**
Decksharing Tooltip enabled. Fix #942

### CODE REVIEW

**Comment by jddeep** [comment]
@ksheremet Please review.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 1]
```diff
@@ -67,14 +72,17 @@ class _DeckSharingState extends State<DeckSharing> {
...
-        title: TextField(
-          controller: _textController,
-          onChanged: (text) {
-            setState(() {});
-          },
-          style: AppStyles.primaryText,
-          decoration: InputDecoration(
-            hintText: AppLocalizations.of(context).emailAddressHint,
+        title: Tooltip(
+          message: '${AppLocalizations.of(context).emailAddressHint}',
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: @jddeep
TextField doesn't need tooltip. Screen readers can read it from the hint.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 2]
```diff
@@ -157,9 +165,12 @@ class _DeckUsersState extends State<DeckUsersWidget> {
             padding: const EdgeInsets.only(left: 8.0, top: 8.0),
             child: Row(
               children: <Widget>[
-                Text(
-                  AppLocalizations.of(context).whoHasAccessLabel,
-                  style: AppStyles.secondaryText,
+                Tooltip(
+                  message: '${AppLocalizations.of(context).whoHasAccessLabel}',
+                  child: Text(
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: @jddeep
Text doesn't need tooltip. Screen readers can read it from the text widget.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 3]
```diff
@@ -194,12 +205,15 @@ class _DeckUsersState extends State<DeckUsersWidget> {
...
-          ? ProgressIndicatorWidget()
-          : Text(
-              displayName,
-              style: AppStyles.primaryText,
-            ),
+      title: Tooltip(
+        message: '$displayName',
+        child: displayName == null
+            ? ProgressIndicatorWidget()
+            : Text(
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: @jddeep
Text doesn't need tooltip. Screen readers can read it from the text widget.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 4]
```diff
@@ -194,12 +205,15 @@ class _DeckUsersState extends State<DeckUsersWidget> {
...
+                displayName,
+                style: AppStyles.primaryText,
+              ),
+      ),
       trailing: DeckAccessDropdownWidget(
         value: accessViewModel.access,
         filter: filter,
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: @jddeep 
Accessibility is not only about Tooltip. Have you had a look on [doc](https://flutter.dev/docs/development/accessibility-and-localization/accessibility)?
Have you tried to install  Accessibility Scanner?

I've tried and it is a problems that I've got in your branch
The following is a list of opportunities to improve the accessibility of Delern DEBUG. Each item corresponds to an outlined area on the attached screenshot.

Text contrast
[0,72][168,240]
The item's text contrast ratio is 2.78. This ratio is based on an estimated foreground color of #FFFFFF and an estimated background color of #4CAF50. Consider increasing this item's text contrast ratio to 3.00 or greater.

Touch target
[48,336][639,474]
This item's height is 46dp. Consider making the height of this touch target 48dp or larger.

Text contrast
[0,570][1080,738]
The item's text contrast ratio is 3.94. This ratio is based on an estimated foreground color of #388E3C and an estimated background color of #FAFAFA. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.

Please consider to fix them and check the results using Accessibility Scanner
screenshot_delern debug_2019-03-07-16_55_10

**Comment by jddeep** [comment]
@ksheremet Then i am removing the Tooltips. Do you suggest to add textscalefactor to all the Text widgets?

**Comment by ksheremet** [comment]
@jddeep not to all, only that were mentioned

**Comment by jddeep** [comment]
@ksheremet Please review.

**Comment by jddeep** [comment]
@dotdoom @ksheremet Please review.

**Comment by jddeep** [comment]
@dotdoom Please kindly check why the CI build is failing?

**Comment by dotdoom** [comment]
Looks like CI is suggesting some formatting changes. Look for "Working directory not clean" message.

**Comment by jddeep** [comment]
@dotdoom @ksheremet Well everything is fine here on my side. I am not getting why this is showing here in CI build.

**Comment by dotdoom** [comment]
The CI diff shows which lines are wrongly formatted.

Have you tried `flutter format` on your side?

**Comment by jddeep** [comment]
I actually don't remember. Will do again and push in that case.

**Comment by jddeep** [comment]
@kshermet @dotodoom please see now.

**Comment by ksheremet** [comment]
@jddeep 
it looks not good

settings_student

Please have a look [here](https://medium.com/flutter-community/a-deep-dive-into-flutters-accessibility-widgets-eb0ef9455bc) and [here](https://stackoverflow.com/questions/52138524/what-is-the-difference-between-fontsize-and-textscalefactor-in-flutter)

@mohak1283 @AbhilashG97 Could you please help @jddeep with the issue

**Comment by jddeep** [comment]
Screenshot_20190312-082225
@ksheremet I have seen into everything. Have a look at this. The only problem remains with the back button `Text contrast ratio` . Any hint on how that can be resolved? Thanks.

**Comment by jddeep** [comment]
Screenshot_20190312-100415
@mohak1283 Please check here.

**Comment by ksheremet** [comment]
@jddeep I checked the code and the problem still persists. Please make sure that you committed your changes

Screenshot_1552387192

**Comment by jddeep** [comment]
@ksheremet No i have not committed the changes

**Comment by jddeep** [comment]
@ksheremet I have committed the changes. Please take a look once.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 5]
```diff
@@ -34,12 +34,16 @@ class _DeckSharingState extends State<DeckSharing> {
   @override
   Widget build(BuildContext context) => Scaffold(
         appBar: AppBar(
+          iconTheme: const IconThemeData(
+            color: Colors.white,
+          ),
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: @jddeep Why did you add `IconThemeData`? It doesn't change anything.

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 6]
```diff
@@ -72,7 +76,12 @@ class _DeckSharingState extends State<DeckSharing> {
           onChanged: (text) {
             setState(() {});
           },
-          style: AppStyles.primaryText,
+          style: TextStyle(
+            fontSize: MediaQuery.of(context).textScaleFactor >=
+                    14.00
+                ? 14.00 * 3.52
+                : 14.00 * 1.50,
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: Why did you try to change textScaleFactor? From the article that I posted in comments before. Have you had a look 
https://medium.com/flutter-community/a-deep-dive-into-flutters-accessibility-widgets-eb0ef9455bc

`One thing to keep in mind is that if you manually set the textScaleFactor, automatic text size calculations from the user’s accessibility settings will be overridden. So, the user’s accessibility settings won’t work anymore. If this causes a large enough problem for the user, then they’re probably just going to uninstall your app.`

What you should do:

1. Please return fontSize back. 

1. From Accessibitily Scanner:  
`The item's text contrast ratio is 2.78. This ratio is based on an estimated foreground color of #FFFFFF and an `

Please change style of hint text (color) that it has good text contrast

File: flutter/lib/views/deck_sharing/deck_sharing.dart [hunk 7]
```diff
@@ -199,6 +208,11 @@ class _DeckUsersState extends State<DeckUsersWidget> {
           : Text(
               displayName,
               style: AppStyles.primaryText,
+              textScaleFactor:
+                  MediaQuery.of(context).textScaleFactor >
+                          1.00
+                      ? 4.52
+                      : 1.50,
```
Comment (ksheremet) [inline flutter/lib/views/deck_sharing/deck_sharing.dart]: Have a look at the comment above. Here is a similar problem
What you should do:

1. Please return fontSize back.
1. Please change style of display name text (color) that it has good text contrast

Comment (mohak1283) [reply flutter/lib/views/deck_sharing/deck_sharing.dart]: Yes @jddeep @ksheremet is absolutely right here. You should try to change the style of display name text here.

**Comment by jddeep** [comment]
@ksheremet Kindly see now, All have been commited.

**Comment by ksheremet** [comment]
@mohak1283 @AbhilashG97 please give a code review. What do you think?

**Comment by mohak1283** [comment]
Hii @jddeep Good job so far 🎉 
Does the text contrast ratio of back button is resolved now by changing it to black color ?

**Comment by jddeep** [comment]
@mohak1283 yes everything is fine now. Thanks.

**Comment by mohak1283** [comment]
Then @ksheremet I think everything is fine now with this PR 👍 
What do you think?

**Comment by jddeep** [comment]
@ksheremet please finalize now. Thanks.

**Comment by ksheremet** [comment]
@jddeep 
App follows Material Design. In AppBar, the color of back arrow hardcoded in black now. Other screens use white colored back arrow - it is inconsistency. Please return color to white by __removing__
```
iconTheme: const IconThemeData(
            color: Colors.white,
          ),
```

To fix the white arrow accessibility I will create a new issue

Everything else is OK. After removing the code provided above I will merge it.

**Comment by jddeep** [comment]
@ksheremet it is done now. Thanks.
